src/plugins/vis_type_timeline/server/index.ts

```diff
@@ -42,11 +42,23 @@ export const config: PluginConfigDescriptor<ConfigSchema> = {
     ui: true,
   },
   deprecations: ({ renameFromRoot }) => [
+    // timeline
     renameFromRoot('timeline_vis.enabled', 'vis_type_timeline.enabled'),
     renameFromRoot('timeline.enabled', 'vis_type_timeline.enabled'),
     renameFromRoot('timeline.graphiteUrls', 'vis_type_timeline.graphiteUrls'),
     renameFromRoot('vis_type_timeline.graphiteUrls', 'vis_type_timeline.graphiteAllowedUrls'),
     renameFromRoot('timeline.ui.enabled', 'vis_type_timeline.ui.enabled', true),
+    // legacy timelion to current timeline
+    renameFromRoot('timelion_vis.enabled', 'vis_type_timeline.enabled'),
+    renameFromRoot('timelion.enabled', 'vis_type_timeline.enabled'),
+    renameFromRoot('timelion.graphiteUrls', 'vis_type_timeline.graphiteUrls'),
+    renameFromRoot('vis_type_timelion.graphiteUrls', 'vis_type_timeline.graphiteAllowedUrls'),
+    renameFromRoot('timelion.ui.enabled', 'vis_type_timeline.ui.enabled', true),
+    // current timelion to current timeline
+    renameFromRoot('vis_type_timelion.enabled', 'vis_type_timeline.enabled'),
+    renameFromRoot('timelion.graphiteUrls', 'vis_type_timeline.graphiteAllowedUrls'),
+    renameFromRoot('vis_type_timelion.graphiteUrls', 'vis_type_timeline.graphiteAllowedUrls'),
+    renameFromRoot('vis_type_timelion.ui.enabled', 'vis_type_timeline.ui.enabled', true),
   ],
```

Comment:
minor issue, do you think it would be more clearer if re-order a bit and put 1 setting together, like this:

```
// timelion.enabled and timelion_vis.enabled deprecation 
renameFromRoot('timelion.enabled', 'vis_type_timeline.enabled'),
renameFromRoot('timelion_vis.enabled', 'vis_type_timeline.enabled'),
renameFromRoot('vis_type_timelion.enabled', 'vis_type_timeline.enabled'),
renameFromRoot('timeline.enabled', 'vis_type_timeline.enabled'),
renameFromRoot('timeline_vis.enabled', 'vis_type_timeline.enabled'),


// timelion.graphiteUrls deprecation
renameFromRoot('timelion.graphiteUrls', 'vis_type_timeline.graphiteUrls'), 
renameFromRoot('timelion.graphiteUrls', 'vis_type_timeline.graphiteAllowedUrls'), 
renameFromRoot('vis_type_timelion.graphiteUrls', 'vis_type_timeline.graphiteAllowedUrls'),
renameFromRoot('timeline.graphiteUrls', 'vis_type_timeline.graphiteUrls'),
renameFromRoot('vis_type_timeline.graphiteUrls', 'vis_type_timeline.graphiteAllowedUrls'),


// timelion.ui.enabled deprecation
renameFromRoot('timelion.ui.enabled', 'vis_type_timeline.ui.enabled', true),
renameFromRoot('vis_type_timelion.ui.enabled', 'vis_type_timeline.ui.enabled', true),
renameFromRoot('timeline.ui.enabled', 'vis_type_timeline.ui.enabled', true),
```

Another question: `renameFromRoot('timelion.graphiteUrls', 'vis_type_timeline.graphiteUrls') `———> we keep this one for backward capacity ???

Reply:
took what you had and modify it because your question got me thinking. I made it so we just want to have the right setting point to the updated settings so I removed that `renameFromRoot('timelion.graphiteUrls', 'vis_type_timeline.graphiteUrls')` and made sure `renameFromRoot('timeline.graphiteUrls', 'vis_type_timeline.graphiteAllowedUrls')`, if we rename it to a previous setting it might actually end up causing a conflict and the incorrect value being picked up.

thanks for the callout!